static/js/stream_edit.js

```diff
@@ -407,61 +407,44 @@ exports.show_settings_for = function (node) {
     show_subscription_settings(sub_settings);
 };
 
-function stream_is_muted_clicked(e) {
-    e.preventDefault();
+function stream_is_muted_changed(e) {
     const sub = get_sub_for_target(e.target);
     if (!sub) {
-        blueslip.error("stream_is_muted_clicked() fails");
+        blueslip.error("stream_is_muted_changed() fails");
         return;
     }
 
     const sub_settings = exports.settings_for_sub(sub);
     const notification_checkboxes = sub_settings.find(".sub_notification_setting");
 
-    subs.toggle_home(sub, `#stream_change_property_status${sub.stream_id}`);
-
-    if (!sub.is_muted) {
-        sub_settings.find(".mute-note").addClass("hide-mute-note");
-        notification_checkboxes.removeClass("muted-sub");
-        notification_checkboxes.find("input[type='checkbox']").prop("disabled", false);
-    } else {
-        sub_settings.find(".mute-note").removeClass("hide-mute-note");
-        notification_checkboxes.addClass("muted-sub");
-        notification_checkboxes.find("input[type='checkbox']").attr("disabled", true);
-    }
+    subs.set_muted(sub, `#stream_change_property_status${sub.stream_id}`, e.target.checked);
+    sub_settings.find(".mute-note").toggleClass("hide-mute-note", !sub.is_muted);
+    notification_checkboxes.toggleClass("muted-sub", sub.is_muted);
+    notification_checkboxes.find("input[type='checkbox']").prop("disabled", sub.is_muted);
 }
 
-exports.stream_setting_clicked = function (e) {
-    if (e.currentTarget.id === "sub_is_muted_setting") {
+exports.stream_setting_changed = function (e, from_notification_settings) {
+    if (e.target.name === "is_muted") {
         return;
     }
-    e.preventDefault();
-    e.stopPropagation();
 
     const sub = get_sub_for_target(e.target);
-    let checkbox = $(e.currentTarget).find(".sub_setting_control");
-    let status_element = "#stream_change_property_status" + sub.stream_id;
-    // sub data is being changed from the notification settings page.
-    if (checkbox.length === 0) {
-        checkbox = $(e.currentTarget);
-        status_element = checkbox.closest(".subsection-parent").find(".alert-notification");
-    }
-    const setting = checkbox.attr("name");
+    let status_element = from_notification_settings
+        ? (status_element = $(e.target).closest(".subsection-parent").find(".alert-notification"))
+        : $("#stream_change_property_status" + sub.stream_id);
```

Comment:
I find this comprehension hard to read.

Reply:
It probably doesn’t help that I wrote it wrong, is this correct version better?

Reply:
Yeah it also likely doesn't help my head has felt like it's in a haze for the last few days.  but yeah this makes sense to me.